roles/ironic-control/tasks/main.yml

```diff
@@ -49,7 +49,7 @@
   tags: db-migrate
 
 - name: sync ironic database
-  command: "{{ ironic.virtualenv }}/bin/ironic-dbsync upgrade"
+  command: "/usr/local/bin/ironic-dbsync upgrade"
```

Comment:
I think you're making a mistake here. When we make use of the `openstack-source` install method, we install into a virtualenv. When we use giftwrap packages, the package is a venv. However we do use the alternatives system to create symlinks in the regular path to these executables. The python env is segregated, but the executable gets exposed in the system path.